File: oauth2-extended-authorization-server/client/src/main/java/com/onlyteo/sandbox/properties/AppSecurityProperties.java
package com.onlyteo.sandbox.properties;

import jakarta.validation.constraints.NotNull;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.ArrayList;
import java.util.List;

@ConfigurationProperties(prefix = "app.security")
public class AppSecurityProperties {

    @NotNull
    private List<String> whitelistedPaths = new ArrayList<>();

    public List<String> getWhitelistedPaths() {
        return whitelistedPaths;
    }

    public void setWhitelistedPaths(List<String> whitelistedPaths) {
        this.whitelistedPaths = whitelistedPaths;
    }

    public String[] getWhitelistedPathsAsArray() {
        return whitelistedPaths.toArray(new String[]{});
    }
}

File: spring-boot-oauth2-client-react/authorization-server/src/main/kotlin/com/onlyteo/sandbox/properties/AppSecurityProperties.kt
package com.onlyteo.sandbox.properties

import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "app.security")
data class AppSecurityProperties(
    @field:Valid @field:NotBlank val whitelistedPaths: List<String>
)